src/NuGet.Clients/NuGet.CommandLine/Commands/ProjectFactory.cs

```diff
@@ -797,6 +802,15 @@ private void ExtractMetadata(Packaging.PackageBuilder builder)
                 {
                     new AssemblyMetadataExtractor(Logger).ExtractMetadata(builder, TargetPath);
                 }
+                catch (FileLoadException flex) when (flex.FileName.EndsWith(".exe", StringComparison.OrdinalIgnoreCase))
```

Comment:
I'm not checking `nuget.exe`, because I rename nuget version 6 to nuget6.exe or something else if I use several nuget.exe versions. I believe other people do same.

Reply:
<img width="59" alt="image" src="https://user-images.githubusercontent.com/8766776/154164206-b3e1eab6-90db-4db0-85b2-e5e852098005.png">

Reply:
Could this exception be thrown if the assembly we're trying to read has issues? 

Can we focus on the exception thrown from https://github.com/NuGet/NuGet.Client/blob/0245b621ffd219fde8a7a6d119af561fb306356f/src/NuGet.Clients/NuGet.CommandLine/Commands/Pack/AssemblyMetadataExtractor.cs#L25 in particular?

The metadata extractor is in nuget.commandline, so we can tweak how we see fit.

Reply:
We can move logic there still we need to rethrow exception here. Otherwise below default `catch (Exception ex)` will swallow the exception/error.

Reply:
So I feel we're making 1 logic into 2 logic. One in `AssemblyMetadataExtractor.cs` and another in here.

Reply:
> The line above where the catch block is now uses `Assembly.GetExecutingAssembly().Location`. Could that be used to get an exact match, rather than `.EndsWith(".exe")`?

Sure, good point. I addressed it now.